ginivision/src/main/java/net/gini/android/vision/analysis/AnalysisFragmentImpl.java

```diff
@@ -112,38 +126,64 @@ public void showError(@NonNull String message, @NonNull String buttonTitle,
     @Override
     public void startScanAnimation() {
         mProgressActivity.setVisibility(View.VISIBLE);
+        mAnalysisMessageTextView.setVisibility(View.VISIBLE);
     }
 
     @Override
     public void stopScanAnimation() {
         mProgressActivity.setVisibility(View.GONE);
+        mAnalysisMessageTextView.setVisibility(View.GONE);
     }
 
     public void onCreate(Bundle savedInstanceState) {
-        forcePortraitOrientationOnPhones(mFragment.getActivity());
+        final Activity activity = mFragment.getActivity();
+        if (activity == null) {
+            return;
+        }
+        forcePortraitOrientationOnPhones(activity);
+        mDocumentRenderer = DocumentRendererFactory.fromDocument(mDocument, activity);
     }
 
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.gv_fragment_analysis, container, false);
         bindViews(view);
-        showDocument();
-        observerViewTree(view);
         return view;
     }
 
     private Handler mHandler = new Handler(Looper.getMainLooper());
 
     public void onDestroy() {
-        mPhoto = null;
+        mImageDocument = null;
         stopScanAnimation();
     }
 
     public void onStart() {
+        mStopped = false;
+        final Activity activity = mFragment.getActivity();
+        if (activity == null) {
+            return;
+        }
+        startScanAnimation();
+        mDocument.loadData(activity, new LoadDataCallback() {
```

Comment:
For camera photos data is returned in the callback immediately.

For files the data is loaded async as it could block the main thread unreasonably long.